admin/src/main/java/io/strimzi/arguments/topic/DescribeTopicCommand.java

```diff
@@ -29,13 +37,38 @@ public Integer call() {
      */
     private Integer describeTopics() {
         try (Admin admin = Admin.create(AdminProperties.adminProperties(this.bootstrapServer))) {
-            admin.describeTopics(List.of(this.getTopicPrefixOrName())).topicNameValues().forEach((key, value) -> {
+            List<KafkaTopicDescription> kafkaTopicDescriptionList = new LinkedList<>();
+
+            // parse kafka response
+            admin.describeTopics(this.getListOfTopicNames()).topicNameValues().forEach((key, value) -> {
                 try {
-                    System.out.println(value.get());
+                    final TopicDescription description = value.get();
+                    final int partitionCount = description.partitions().size();
+                    // as each partition has the same size, replica count is deduced from first partition
+                    final int replicaCount = description.partitions().get(0).replicas().size();
+                    KafkaTopicDescription kafkaTopicDescription = new KafkaTopicDescription(key, partitionCount, replicaCount);
+                    kafkaTopicDescriptionList.add(kafkaTopicDescription);
```

Comment:
Did you thought about having this inside the `KafkaTopicDescription`? It would be easier to test + it could be then used on more places (if needed).

so the code then could look like

```
kafkaTopicDescriptionList.add(KafkaTopicDescription.parse(value));
```
and I know that you used `record` just for storing the value.